Add corresponding check with cmake_minimum_required for ASan build

If one tries to build Tarantool with ASan using old CMake version, he sees the following error:
```bash
$ CC=clang CXX=clang++ cmake . -DENABLE_WERROR=ON -DENABLE_ASAN=ON -DENABLE_UB_SANITIZER=ON -DENABLE_FUZZER=ON
...
CMake Error at cmake/compiler.cmake:284 (string):
  string does not recognize sub-command JOIN
Call Stack (most recent call first):
  src/CMakeLists.txt:4 (enable_tnt_compile_flags)
```

CMake version:
```
$ cmake --version
cmake version 3.1.3
``` 

According to [this](https://cmake.org/cmake/help/latest/command/string.html#join) section of CMake documentation `string(JOIN, ...)` was introduced in CMake version 3.10.

--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -1,3 +1,5 @@
+include(cmake/utils.cmake)
+
 #
 # Check if the same compile family is used for both C and CXX
 #
@@ -234,7 +236,7 @@ macro(enable_tnt_compile_flags)
         if (NOT CMAKE_COMPILER_IS_CLANG)
             message(FATAL_ERROR "Undefined behaviour sanitizer only available for clang")
         else()
-            string(JOIN "," SANITIZE_FLAGS
+            string_join("," SANITIZE_FLAGS
                 alignment bool bounds builtin enum float-cast-overflow
                 float-divide-by-zero function integer-divide-by-zero return
                 shift unreachable vla-bound

--- a/cmake/utils.cmake
+++ b/cmake/utils.cmake
@@ -145,3 +145,21 @@ function(list_add_prefix
     endforeach()
     set(${list_out} ${result} PARENT_SCOPE)
 endfunction()
+
+# string() has JOIN option but only since 3.12.
+function(string_join
+    glue
+    string_out
+)
+    set(result "")
+    set(is_first_item TRUE)
+    foreach(item IN LISTS ARGN)
+        if(is_first_item)
+            set(result "${item}")
+            set(is_first_item FALSE)
+        else()
+            set(result "${result}${glue}${item}")
+        endif()
+    endforeach()
+    set(${string_out} ${result} PARENT_SCOPE)
+endfunction()

